### PR TITLE
Enhance configuration usage

### LINKED ISSUE #861: --cluster-name flag does not accept "-" character
# --cluster-name flag does not accept "-" character, I did not test it with any other character, because normally one would use the "-" character

```
+ kubescape scan framework nsa --verbose --cluster-name=my-cluster1-1-0
{"level":"info","ts":"2022-10-04T15:39:21Z","msg":"Kubescape scanner starting"}
{"level":"fatal","ts":"2022-10-04T15:39:21Z","msg":"custom cluster name cannot contain special characters"}
```

# Environment
OS:  Ubuntu 22.04 LTS

Your current version is:
v2.0.172

# Steps To Reproduce
1. Run this
kubescape scan framework nsa --verbose --cluster-name=my-cluster1-1-0

# Expected behavior
Scan should run

# Actual Behavior
Scan does not run

# Additional context

### PR DESCRIPTION
**Pull request by dwertent**
## Describe your changes
* Split build workflows
* Change the user name to `ks`
* Load URLs from config
* Adding diagrams to README

## This PR fixes:

* Resolved #861 

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)**

### CODE REVIEW

File: core/pkg/hostsensorutils/hostsensor.yaml [hunk 1]
```diff
@@ -28,12 +28,15 @@ spec:
...
+      - key: node-role.kubernetes.io/control-plane
+        operator: Exists
+        effect: NoSchedule
       - key: node-role.kubernetes.io/master
         operator: Exists
         effect: NoSchedule
       containers:
       - name: host-sensor
-        image: quay.io/kubescape/host-scanner:latest
+        image: quay.io/kubescape/host-scanner:v1.0.28
```
Comment (amirmalka) [inline core/pkg/hostsensorutils/hostsensor.yaml:39]: just making sure it's intentional...

File: core/pkg/resourcehandler/k8sresources.go [hunk 2]
```diff
@@ -79,9 +79,14 @@ func (k8sHandler *K8sResourceHandler) GetResources(sessionObj *cautils.OPASessio
 		sessionObj.SetNumberOfWorkerNodes(numberOfWorkerNodes)
 	}
 
+	cautils.StopSpinner()
+	logger.L().Success("Accessed to Kubernetes objects")
+
 	imgVulnResources := cautils.MapImageVulnResources(ksResourceMap)
 	// check that controls use image vulnerability resources
-	if len(imgVulnResources) > 0 {
+	if false { //len(imgVulnResources) > 0 {
```
Comment (amirmalka) [inline core/pkg/resourcehandler/k8sresources.go:87]: ?

File: httphandler/build.py [hunk 3]
```diff
@@ -6,60 +6,63 @@
...
-    buildDir = "build/"
+def get_build_dir():
+    current_platform = platform.system()
+    build_dir = "build/"
 
-    if currentPlatform == "Windows": return os.path.join(buildDir, "windows-latest") 
-    if currentPlatform == "Linux": return os.path.join(buildDir, "ubuntu-latest")  
-    if currentPlatform == "Darwin": return os.path.join(buildDir, "macos-latest")  
-    raise OSError("Platform %s is not supported!" % (currentPlatform))
+    if current_platform == "Windows": build_dir += "windows-latest"
```
Comment (amirmalka) [inline httphandler/build.py:19]: I believe that previous code was more platform generic when using `os.path.join`; As now the build_dir has a hardcoded path separator (which is incorrect for Windows)

File: httphandler/build.py [hunk 4]
```diff
@@ -6,60 +6,63 @@
...
+    elif current_platform == "Linux": build_dir += "ubuntu-latest"
+    elif current_platform == "Darwin": build_dir += "macos-latest"
+    else: raise OSError("Platform %s is not supported!" % (current_platform))
 
-def getPackageName():
-    packageName = "kubescape"
-    # if platform.system() == "Windows": packageName += ".exe"
```
Comment (amirmalka) [inline httphandler/build.py:26]: Same here.. `packageName` is used later for the output filename. which should be ending with `.exe` for Windows..

File: README.md [hunk 5]
```diff
@@ -51,6 +51,13 @@ kubescape scan --submit --enable-host-scan --verbose
 
 </br>
 
+## Architecture in short
+### CLI
+<img src="docs/ks-cli-arch.png" width="300">
+
+### Operator
+<img src="docs/ks-operator-arch.png" width="300">
```
Comment (amirmalka) [inline README.md:59]: I think would be nicer to add the diagrams as mermaid instead...

Comment (dwertent) [reply README.md:59]: I agree, Ben insisted on using an image

**Review by amirmalka — APPROVED** [review]

**Pull request merged**